WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentsViewModel.kt

```diff
@@ -154,30 +176,37 @@ class NewSiteCreationSegmentsViewModel
             items.add(segment)
         }
     }
+}
 
-    data class UiState(
-        val showError: Boolean,
-        val showContent: Boolean,
-        val items: List<ItemUiState>
-    )
+interface SegmentsUiState
+data class SegmentsContentUiState(val items: List<SegmentsItemUiState>) : SegmentsUiState
+data class SegmentsErrorUiState constructor(val titleResId: Int, val subtitleResId: Int? = null) : SegmentsUiState {
+    companion object {
+        fun createGenericErrorUiState() = SegmentsErrorUiState(
+                R.string.site_creation_error_generic_title,
+                R.string.site_creation_error_generic_subtitle
+        )
 
-    sealed class ItemUiState {
-        object HeaderUiState : ItemUiState() {
-            val titleResId: Int = R.string.site_creation_segments_title
-            val subtitleResId: Int = R.string.site_creation_segments_subtitle
-        }
+        fun createConnectionErrorUiState() = SegmentsErrorUiState(R.string.site_creation_error_connection_title)
```

Comment:
Can we change this to a `val`?

```suggestion
        val createConnectionErrorUiState = SegmentsErrorUiState(R.string.site_creation_error_connection_title)
```

Reply:
👍 -> I've used the proposed solution from the other comment.